ipaserver/install/servntpconf.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+from __future__ import absolute_import
+
+from logging import getLogger
+
+from ipaplatform.paths import paths
+from ipapython import ipautil
+from ipalib.install import sysrestore
+from ipaserver.install import service
+from ipapython import ntpmethods
+from ipapython.ntpmethods import detect_time_server
+
+logger = getLogger(__name__)
+
+
+class BaseNTPServer(service.Service):
+    timeout = 15
+
+    def __init__(self, service_name, ntp_confile=None, fstore=None,
+                 ntp_bin=None, ntp_servers=None, ntp_pool=None,
+                 sstore=None, flag=None, opts=None):
+        super(BaseNTPServer, self).__init__(
+            service_name=service_name,
+            fstore=fstore,
+            service_desc="NTP daemon",
+            sstore=sstore,
+        )
+
+        self.ntp_confile = ntp_confile
+        self.ntp_servers = ntp_servers
+        self.ntp_pool = ntp_pool
+        self.ntp_bin = ntp_bin
+        self.flag = flag
+        self.opts = opts
+
+        self.args = [paths.BIN_TIMEOUT, str(self.timeout),
+                     self.ntp_bin, self.flag]
+
+        if not fstore:
+            self.fstore = sysrestore.FileStore(paths.SYSRESTORE)
+
+    def __configure_ntp(self):
+
+        logger.debug("Backing up %s", self.ntp_confile)
+        ntpmethods.backup_config(self.ntp_confile, self.fstore)
+
+        logger.debug("Configuring %s", detect_time_server())
+
+        enabled = ntpmethods.service_command()['api'].is_enabled()
+        running = ntpmethods.service_command()['api'].is_running()
+
+        if self.sstore:
+            self.sstore.backup_state(ntpmethods.service_command()['service'],
+                                     'enabled', enabled)
+            self.sstore.backup_state(ntpmethods.service_command()['service'],
+                                     'running', running)
+
+        if not self.ntp_servers and not self.ntp_pool:
+            self.ntp_pool = "pool.ntp.org"
+
+        config_content = ntpmethods.set_config(
+            path=self.ntp_confile,
+            pool=self.ntp_pool,
+            servers=self.ntp_servers,
+            opts=self.opts,
+        )
+
+        logger.debug("Writing configuration to %s", self.ntp_confile)
+
+        ntpmethods.service_command()['api'].stop()
+        ntpmethods.write_config(self.ntp_confile, config_content)
+
+    def __start_sync(self):
+        logger.debug("Synchronizing time with %s", detect_time_server())
+        ipautil.run(self.args)
+
+    def sync_time(self):
+        self.step("stopping %s" % self.service_name, self.stop)
+        self.step("writing configuration", self.__configure_ntp)
+        self.step("configuring %s to start on boot"
+                  % self.service_name, self.enable)
+        self.step("synchronization time with %s"
```

Comment:
In this step installation is broken and after installation i see:
```
# systemctl status chronyd
● chronyd.service - NTP client/server
   Loaded: loaded (/usr/lib/systemd/system/chronyd.service; enabled; vendor preset: enabled)
   Active: inactive (dead) 
```
As you can see in installation output:
```
  [4/5]: synchronization time with chrony
  [error] CalledProcessError: CalledProcessError(Command ['/usr/bin/timeout', '15', '/usr/sbin/chronyd', '-q'] returned non-zero exit status 1: '2019-02-04T12:52:44Z chronyd version 3.3 starting (+CMDMON +NTP +REFCLOCK +RTC +PRIVDROP +SCFILTER +SIGND +ASYNCDNS +SECHASH +IPV6 +DEBUG)\n2019-02-04T12:52:44Z Frequency -1.553 +/- 0.011 ppm read from /var/lib/chrony/drift\n2019-02-04T12:52:44Z Using right/UTC timezone to obtain leap second data\n2019-02-04T12:52:55Z No suitable source for synchronisation\n2019-02-04T12:52:55Z chronyd exiting\n')
Warning: IPA was unable to sync time with chrony!
         Time synchronization is required for IPA to work correctly
Configuring directory server (dirsrv). Estimated time: 30 seconds
  [1/44]: creating directory server instance
```
see __start_sync()

Reply:
self.ntp_confile should not be empty. Can I see the log file?

Reply:
Of course:
https://pastebin.com/KrBMvFkE

Reply:
Checked on several different systems, could not reproduce the error. Chrony synchronization works.

Reply:
Thanks for feedback, I will try again tomorrow.